### PR TITLE
Create TestConsensusNode

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- missing a generic method to create a node for consensus integration tests
  
## Proposed Changes
- create testConsensusNode to be used further in integration tests

## Testing procedure
- N/A, integration tests only

### CODE REVIEW

File: integrationTests/testConsensusNode.go [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+	ShardCoordinator sharding.Coordinator
+	ChainHandler     data.ChainHandler
+	BlockProcessor   *mock.BlockProcessorMock
+	ResolverFinder   dataRetriever.ResolversFinder
+	AccountsDB       *state.AccountsDB
+	NodeKeys         TestKeyPair
+}
+
+// NewTestConsensusNode returns a new TestConsensusNode
+func NewTestConsensusNode(
```
Comment (iulianpascalau) [inline integrationTests/testConsensusNode.go:71]: for simplicity, we might have removed maxShards and nodeShardId as the consensus is always tested intra shard. (could have been hardcoded through some consts to 1 and 0)

File: integrationTests/testConsensusNode.go [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+		node.WithValidatorSignatureSize(signatureSize),
+		node.WithPublicKeySize(publicKeySize),
+	)
+
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+}
+
+func (tcn *TestConsensusNode) initShardCoordinator(
```
Comment (iulianpascalau) [inline integrationTests/testConsensusNode.go:286]: this is actually initNodesCoordinator

File: integrationTests/testConsensusNode.go [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+	tcn.NodesCoordinator, _ = nodesCoordinator.NewIndexHashedNodesCoordinator(argumentsNodesCoordinator)
+}
+
+func (tcn *TestConsensusNode) initBlockChain(hasher hashing.Hasher) {
+	if tcn.ShardCoordinator.SelfId() == core.MetachainShardId {
+		tcn.ChainHandler = CreateMetaChain()
+	} else {
+		tcn.ChainHandler = CreateShardChain()
+	}
+	//_ = tcn.ChainHandler.SetGenesisHeader(&dataBlock.Header{})
```
Comment (iulianpascalau) [inline integrationTests/testConsensusNode.go:327]: commented code

File: integrationTests/testConsensusNode.go [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+
+	_ = tcn.ChainHandler.SetGenesisHeader(header)
+	hdrMarshalized, _ := TestMarshalizer.Marshal(header)
+	tcn.ChainHandler.SetGenesisHeaderHash(hasher.Compute(string(hdrMarshalized)))
+}
+
+func (tcn *TestConsensusNode) initBlockProcessor() {
+	tcn.BlockProcessor = &mock.BlockProcessorMock{
+		ProcessBlockCalled: func(header data.HeaderHandler, body data.BodyHandler, haveTime func() time.Duration) error {
+			_ = tcn.ChainHandler.SetCurrentBlockHeaderAndRootHash(header, header.GetRootHash())
```
Comment (iulianpascalau) [inline integrationTests/testConsensusNode.go:348]: in the real component this is actually done inside the CommitBlock part. We should remove it from here

File: integrationTests/testConsensusNode.go [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+	args := trie.NewTrieStorageManagerArgs{
+		MainStorer:             CreateMemUnit(),
+		CheckpointsStorer:      CreateMemUnit(),
+		Marshalizer:            TestMarshaller,
+		Hasher:                 hasher,
+		GeneralConfig:          generalCfg,
+		CheckpointHashesHolder: hashesHolder.NewCheckpointHashesHolder(10000000, uint64(hasher.Size())),
+		IdleProvider:           &testscommon.ProcessStatusHandlerStub{},
+	}
+	trieStorage, _ := trie.NewTrieStorageManager(args)
```
Comment (ssd04) [inline integrationTests/testConsensusNode.go:413]: here i think we can refactor this to use the `CreateTrieStorageManager` function:
```suggestion
	storer := CreateMemUnit()
	trieStorage, _ := CreateTrieStorageManager(storer)
```

Comment (ssd04) [reply integrationTests/testConsensusNode.go:413]: i think here we can refactor to use `CreateTrieStorageManager` function

**Review by iulianpascalau — DISMISSED** [review]

**Review by ssd04 — DISMISSED** [review]

File: integrationTests/consensus/consensus_test.go [hunk 6]
```diff
@@ -17,7 +18,13 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-const consensusTimeBetweenRounds = time.Second
+const (
+	consensusTimeBetweenRounds = time.Second
+	blsConsensusType           = "bls"
+)
+
+var p2pBootstrapDelay = time.Second * 5
```
Comment (bogdan-rosianu) [inline integrationTests/consensus/consensus_test.go:26]: could have grouped the variables as well in a single block

File: integrationTests/testConsensusNode.go [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+
+	tcn.initNodesCoordinator(consensusSize, testHasher, epochStartRegistrationHandler, eligibleMap, waitingMap, pkBytes, consensusCache)
+	tcn.Messenger = CreateMessengerWithNoDiscovery()
+	tcn.initBlockChain(testHasher)
+	tcn.initBlockProcessor()
+
+	startTime := time.Now().Unix()
+
+	singlesigner := &ed25519SingleSig.Ed25519Signer{}
+	singleBlsSigner := &mclsinglesig.BlsSingleSigner{}
```
Comment (bogdan-rosianu) [inline integrationTests/testConsensusNode.go:149]: wondering why we don't have these 2 signers exported as constants in `integrationTests` package

Comment (sstanculeanu) [reply integrationTests/testConsensusNode.go]: extracted and refactored multiple tests

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**